### PR TITLE
Use jekyll as the build engine for template sites

### PR DESCRIPTION
**Pull request by jmhooper**
The template sites are all jekyll sites, but default build engine for sites is the "static" build engine. Because of this, the "jekyll" option needs to be set explicitly in the SiteCreator. This commit does that.

ref #722

### CODE REVIEW

File: test/api/unit/services/SiteCreator.test.js [hunk 1]
```diff
@@ -281,6 +281,23 @@ describe("SiteCreator", () => {
         }).catch(done)
       })
 
+      it("should use jekyll as the build engine", done => {
+        const siteParams = {
+          owner: crypto.randomBytes(3).toString("hex"),
+          repository: crypto.randomBytes(3).toString("hex"),
+          template: "microsite",
```
Comment (wslack) [inline test/api/unit/services/SiteCreator.test.js]: Why only the microsite here? Shouldn't this apply to all sites?

**Comment by wslack** [comment]
Well, that's no good. Suggest merging this into prod when fixed given the importance of the bug.

File: test/api/unit/services/SiteCreator.test.js (hunk 1 above)
Comment (jmhooper) [inline test/api/unit/services/SiteCreator.test.js]: This is a test case. It is set for all sites in `SiteCreator.js` above.

File: api/services/SiteCreator.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
 
 const createSiteFromTemplate = ({ siteParams, user, template }) => {
   let site = Site.build(siteParams)
+  site.engine = "jekyll"
```
Comment (brandocalrissian) [inline api/services/SiteCreator.js]: Would it be better conceptually to default all sites to use the jekyll engine?

Comment (brandocalrissian) [inline api/services/SiteCreator.js]: No, then we'd be coupling the site creation code with our assumptions around how templated sites are created. Not good.

**Pull request merged**